task/thermal/src/control.rs

```diff
@@ -927,22 +927,27 @@ impl<'a> ThermalControl<'a> {
     /// In cases where dynamic inputs are not present (i.e. they are `None` in
     /// the array), the iterator will skip that entire tuple.
     fn zip_temperatures<'b, T>(
-        values: &'b [T],
-        (static_inputs, dynamic_inputs): (
-            &'b [InputChannel],
-            &'b [Option<DynamicInputChannel>],
-        ),
-    ) -> impl Iterator<Item = (&'b T, ThermalProperties)> {
+        values: &'b [T; TEMPERATURE_ARRAY_SIZE],
+        static_inputs: &'b [InputChannel; bsp::NUM_TEMPERATURE_INPUTS],
+        dynamic_inputs: &'b [Option<DynamicInputChannel>;
+                bsp::NUM_DYNAMIC_TEMPERATURE_INPUTS],
+        dynamic_sensors: &'b [SensorId; bsp::NUM_DYNAMIC_TEMPERATURE_INPUTS],
```

Comment:
Is there a better way to describe these than just `dynamic_inputs` and `dynamic_sensors`? When I was looking at the call site below these are both `dynamic_inputs` which confused me slightly

```
                for (sensor_id, v, model) in Self::zip_temperatures(
                    values,
                    self.bsp.inputs,
                    &self.dynamic_inputs,
                    self.bsp.dynamic_inputs,
                )
```